admin-auth/src/test/kotlin/com/epam/drill/admin/auth/UserRepositoryImplTest.kt

```diff
@@ -15,19 +15,234 @@
+        insertRandomUsers(1..1) {
+            it[username] = "FooBar"
+        }
+
+        assertFails {
+            repository.create(
+                UserEntity(
+                    username = "foobar", passwordHash = "hash", role = "USER"
+                )
+            )
+        }
+    }
+
+    @Test
+    fun `given existing id, update must change user`() = withTransaction {
+        insertRandomUsers(1..10)
+        val id = insertRandomUsers(11..11) {
+            it[username] = "foo"
+            it[passwordHash] = "hash1"
+            it[role] = "USER"
+        }.first()
+        insertRandomUsers(12..20)
+
+        repository.update(
+            UserEntity(
+                id = id, username = "bar", passwordHash = "hash2", role = "ADMIN", deleted = true, blocked = true
+            )
+        )
+
+        assertEquals(1, UserTable.select { UserTable.username eq "bar" }.count())
+        UserTable.select { UserTable.username eq "bar" }.first().let {
+            assertEquals("bar", it[UserTable.username])
+            assertEquals("hash2", it[UserTable.passwordHash])
+            assertEquals("ADMIN", it[UserTable.role])
+            assertTrue(it[UserTable.deleted])
+            assertTrue(it[UserTable.blocked])
+        }
     }
 
     @Test
-    fun `given`() {
+    fun `findAllNotDeleted must return all not deleted users`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..15) {
+            it[deleted] = true //insert 5 deleted users
+        }
+        insertRandomUsers(16..20)
+
+        val users = repository.findAllNotDeleted()
+
+        assertEquals(20 - 5, users.size)
+    }
+
+    @Test
+    fun `given existing username, findByUsername must return user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
+
+        assertNotNull(user)
+        assertEquals("foobar", user.username)
+    }
+
+    @Test
+    fun `findByUsername must return even blocked user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+            it[blocked] = true
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
 
+        assertNotNull(user)
+        assertEquals("foobar", user.username)
+        assertTrue(user.blocked)
     }
 
+    @Test
+    fun `findByUsername mustn't return deleted user`() = withTransaction {
+        insertRandomUsers(1..10)
+        insertRandomUsers(11..11) {
+            it[username] = "foobar"
+            it[deleted] = true
+        }
+        insertRandomUsers(12..20)
+
+        val user = repository.findByUsername("foobar")
+
+        assertNull(user)
+    }
+
+    @Test
+    fun `given case insensitive username, findByUsername must return user`() = withTransaction {
```

Comment:
`findByUsername must be case insensitive`
"case insensitivity" is a characteristic of algorithm/function, not the data